fava/static/javascript/ingest.js

```diff
@@ -0,0 +1,120 @@
+import { $, $$, handleJSON } from './helpers';
+import e from './events';
+
+function initRow(row) {
+  row.querySelectorAll('.action input').forEach((el) => {
+    el.addEventListener('click', () => {
+      row.className = `ingest-row ${el.value}`; // eslint-disable-line no-param-reassign
+    });
+  });
+}
+
+function submitIngestForm() {
+  const form = $('.ingest-extract');
+  const jsonData = { entries: [] };
+
+  form.querySelectorAll('.ingest-row.import .transaction-form').forEach((transaction) => {
```

Comment:
This code is very repetitive. Can't we iterate over all `.ingest-row.import`, for each of them iterate over all `input`s and then set `entryData[input.name] = input.value`. The "type" can also be determined from the HTML, so we'd only need to special-case postings, but otherwise we could handle transactions, balances, and notes in one go.